server/libs/utils/utils.go

```diff
@@ -479,10 +481,35 @@ func IsNil(i interface{}) bool {
 
 // EscapeJSONString is used to escape special characters in JSON strings
 func EscapeJSONString(value string) string {
-	replacer := strings.NewReplacer(
-		`\`, `\\`,
-		`"`, `\"`,
-		"\n", `\n`,
-	)
-	return replacer.Replace(value)
+	var buf bytes.Buffer
+	for i := 0; i < len(s); {
+		r, size := utf8.DecodeRuneInString(s[i:])
```

Comment:
我觉得 参照 https://github.com/Stenway/RSV-Challenge/blob/4948917321f9ae2cef0beb73be41dc959d07bd0b/Go/rsv.go#L244 转换即可，
不用校验所有字符都是合法的utf-8, 可以有非法的utf-8字符，只要也不破坏json格式就行，影响的只是显示的时候是乱码。
如果严格校验的话，影响解析性能，收益只是将乱码展示为 \uxxxx。
----------------------------------------------------------------------------
I think it can be converted by referring to https://github.com/Stenway/RSV-Challenge/blob/4948917321f9ae2cef0beb73be41dc959d07bd0b/Go/rsv.go#L244.
There is no need to verify that all characters are legal UTF-8, there can be illegal UTF-8 characters, as long as the json format is not destroyed, the only impact is that the display will be garbled.
If the verification is strict, the parsing performance will be affected, and the only benefit will be that the garbled code will be displayed as \uxxxx.